core/consensus/slots/src/slots.rs

```diff
@@ -81,6 +82,8 @@ pub struct SlotInfo {
 	pub inherent_data: InherentData,
 	/// Slot duration.
 	pub duration: u64,
+	/// Epoch.
+	pub epoch: u64,
```

Comment:
Do we need to track this here? Why not have just BABE track this based on slot number?

Reply:
I do not think we do.  If I recall correctly, this was done for convenience, but I can fix this.